src/sentry/web/client_config.py

```diff
@@ -153,7 +153,8 @@ def get_client_config(request=None):
         "statuspage": _get_statuspage(),
         "messages": [{"message": msg.message, "level": msg.tags} for msg in messages],
         "apmSampling": float(settings.SENTRY_FRONTEND_APM_SAMPLING or 0),
-        "isOnPremise": settings.SENTRY_ONPREMISE,
+        "isOnPremise": settings.SENTRY_SELF_HOSTED,  # deprecated, use ...
+        "isSelfHosted": settings.SENTRY_SELF_HOSTED,  # ... this instead
```

Comment:
Second-riskiest part. `getsentry` does consume the `isOnPremise` config key in some places. Third-party JavaScript (plugins?) might as well. Preserve the key!

Reply:
Will we be removing `isOnPremise` after `getsentry` is updated?

Reply:
I'd imagine it shouldn't affect them, if they are customizing their install, they shouldn't have a need to differentiate between the two? But otherwise sounds good

Reply:
I mean the default is true. Nobody except us in SaaS should need to change the default. Also nobody in the wild building their own plugins or whatever should need to vary their behavior on whether this config is true or false, realistically they would just always deliver whatever behavior they're building for their custom app/plugin. I think I'm convincing myself that we can remove this sooner rather than later.

Reply:
Noted on https://github.com/getsentry/self-hosted/issues/796. 👍